src/resource/mock.ts

```diff
@@ -0,0 +1,127 @@
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  Url,
+  UrlString,
+  SolidDataset,
+  WithResourceInfo,
+  internal_toIriString,
+} from "../interfaces";
+import { getSolidDataset, createSolidDataset } from "./solidDataset";
+import { getFile } from "./nonRdfData";
+
+type Unpromisify<T> = T extends Promise<infer R> ? R : T;
+
+/**
+ * Function for use in unit tests to mock a persisted [[SolidDataset]].
+ *
+ * Warning: do not use this function in actual production code.
+ * This function initialises a new SolidDataset but, as opposed to [[createSolidDataset]],
+ * it adds metadata that would also have been present had an empty SolidDataset been fetched from
+ * the given URL. This is useful to mock a SolidDataset in tests of code that call e.g.
+ * [[getSourceUrl]].
+ *
+ * @param url The URL from which the mocked SolidDataset pretends to be retrieved.
+ * @returns A mocked SolidDataset.
+ * @since Not released yet.
+ */
+export function mockSolidDatasetFrom(
+  url: Url | UrlString
+): Unpromisify<ReturnType<typeof getSolidDataset>> {
+  const solidDataset = createSolidDataset();
+  const solidDatasetWithResourceInfo: SolidDataset &
+    WithResourceInfo = Object.assign(solidDataset, {
+    internal_resourceInfo: {
+      sourceIri: internal_toIriString(url),
+      isRawData: false,
+      contentType: "text/turtle",
+    },
+  });
+
+  return solidDatasetWithResourceInfo;
+}
+
+/**
+ * Function for use in unit tests to mock a persisted [[SolidDataset]] representing a Container.
+ *
+ * Warning: do not use this function in actual production code.
+ * This function initialises a new SolidDataset but, as opposed to [[createSolidDataset]],
+ * it adds metadata that would also have been present had an empty SolidDataset been fetched from
+ * the given URL, and was a Container. This is useful to mock a SolidDataset in tests of code that
+ * call e.g. [[isContainer]].
+ *
+ * @param url The URL from which the mocked Container pretends to be retrieved — this should end in a slash.
+ * @returns A mocked SolidDataset.
+ * @since Not released yet.
+ */
+export function mockContainerFrom(
+  url: Url | UrlString
+): Unpromisify<ReturnType<typeof getSolidDataset>> {
+  const sourceIri = internal_toIriString(url);
+  if (!sourceIri.endsWith("/")) {
+    throw new Error(
+      "A Container's URL should end in a slash. Please update your tests."
+    );
+  }
+
+  const solidDataset = createSolidDataset();
+  const solidDatasetWithResourceInfo: SolidDataset &
+    WithResourceInfo = Object.assign(solidDataset, {
+    internal_resourceInfo: {
+      sourceIri: sourceIri,
+      isRawData: false,
+      contentType: "text/turtle",
+    },
+  });
+
+  return solidDatasetWithResourceInfo;
```

Comment:
This could return `mockSolidDatasetFrom(url)`

Reply:
Now why did I not do this the first time...? Maybe because we might want to check a header or something for this down the road, but that's something we can resolve at that time.

f65305ca4c70ebd473cb937bd492b4f7770f7d50